### ISSUE TITLE
feature request: clone "xyz extension" documentation to chaining methods

### ISSUE THREAD

**Issue by Lootwig** [gskinner/flutter_animate#125]
I am a fan of the `.animate().fade().slide()` API!

When aiming to achieve a specific effect and just browsing the available methods in the autocompletion overlay, frequently I stumble upon something like this:

```dart
...
fadeIn()
slideX()
slideY()
soundsExactlyLikeWhatYouNeed()
move()
hide()
...
```

then I select the effect that just sounds like what I need, and its docstring is... *drumroll*:

`adds a SoundsJustLikeWhatYouNeed() extension.`

which is pretty much just repeating the method name, but using more words 😁

I've seen the `@macro` annotation used a lot in the flutter source code, which seems to allow embedding some already written doc into another piece of the code without duplicating it - could this be used to elaborate on how to _use_ every extension in their corresponding chaining method?

**Comment by gskinner** [comment]
Right. I struggle with this a bit, because I'm trying to keep the "cost" of adding simple shortcuts down (ie. minimize documentation / maintenance burden), but I totally get your point.

Perhaps the use of a macro, with a very minimal amount of additional explanation could work. I'll maybe play with it on one of the simpler effects to see if I can come up with a simple template that doesn't add a ton of extra work.

If you'd like to take a stab at this and share (just worry about a single effect to start), I'd be happy to let you. :)

**Comment by gskinner** [comment]
Started on this. I don't think there's a reasonable way to get a decent description by using macros, so I'm just doing the grunt work of adding decent descriptions to each method individually following a common formula.

**Comment by gskinner** [comment]
Addressed in a11832679995bd247d05cb190a948b1795b29567

Closing. Feel free to reopen or comment further as appropriate.
